openlibrary/catalog/add_book/test_add_book.py

```diff
@@ -56,6 +62,21 @@ def test_build_query(add_languages):
 
     pytest.raises(InvalidLanguage, build_query, {'languages': ['wtf']})
 
+def test_isbns_from_record():
+    rec = {'title': 'test', 'isbn_13': ['9780190906764'], 'isbn_10': ['0190906766']}
+    result = isbns_from_record(rec)
+    assert isinstance(result, list)
+    assert '9780190906764' in result
+    assert '0190906766' in result
+    assert len(result) == 2
+
+def test_editions_matched(mock_site):
+    rec = {'title': 'test', 'isbn_13': ['9780190906764'], 'isbn_10': ['0190906766']}
+    isbns = isbns_from_record(rec)
+    result = editions_matched(rec, 'isbn_', isbns)
+    #TODO: does a thing search with key = 'isbn_' work?
```

Comment:
Yes, **confirmed** key `isbn_` will return a matching record on either `isbn_10` or `isbn_13` metadata fields!

Reply:
unfortunately I can't get the test to pass with a match using `isbn_`, something is different. Needs further testing

Reply:
Looks like this works on the live site, but not with the test `mock_site` see note at https://github.com/internetarchive/openlibrary/blob/75452cf6f0e3c23e1030395c93e232686ebaa417/openlibrary/records/tests/test_functions.py#L232

Reply:
I have improved `mock_site` so it behaves like the live site in terms of supporting queries by `isbn_` -- not finding any matches on `isbn_` was skipping a whole lot of code paths in and after `openlibrary.catalog.add_book.editions_matched()` which was always returning `[]`